airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/avro_parser.py

```diff
@@ -19,7 +41,68 @@ async def infer_schema(
+        if isinstance(avro_field, Mapping) and avro_field["type"] == "record":
+            return {
+                "type": "object",
+                "properties": {
+                    object_field["name"]: {**AvroParser._convert_avro_type_to_json(object_field["name"], object_field["type"])}
+                    for object_field in avro_field["fields"]
+                },
+            }
+        elif isinstance(avro_field, Mapping) and avro_field["type"] == "array":
+            if "items" not in avro_field:
+                raise ValueError(f"{field_name} array type does not have a required field items")
+            return {"type": "array", "items": {**AvroParser._convert_avro_type_to_json("", avro_field["items"])}}
+        elif isinstance(avro_field, Mapping) and avro_field["type"] == "enum":
+            if "symbols" not in avro_field:
+                raise ValueError(f"{field_name} enum type does not have a required field symbols")
+            if "name" not in avro_field:
+                raise ValueError(f"{field_name} enum type does not have a required field name")
+            return {"type": "string", "enum": avro_field["symbols"]}
+        elif isinstance(avro_field, Mapping) and avro_field["type"] == "map":
+            if "values" not in avro_field:
+                raise ValueError(f"{field_name} map type does not have a required field values")
+            return {"type": "object", "additionalProperties": {**AvroParser._convert_avro_type_to_json("", avro_field["values"])}}
+        elif isinstance(avro_field, Mapping) and avro_field["type"] == "fixed" and avro_field.get("logicalType") != "duration":
+            if "size" not in avro_field:
+                raise ValueError(f"{field_name} fixed type does not have a required field size")
+            if not isinstance(avro_field["size"], int):
+                raise ValueError(f"{field_name} fixed type size value is not an integer")
+            return {
+                "type": "string",
+                "pattern": f"^[0-9A-Fa-f]{{{avro_field['size'] * 2}}}$",
+            }
+        elif isinstance(avro_field, Mapping) and avro_field.get("logicalType") == "decimal":
+            if "precision" not in avro_field:
+                raise ValueError(f"{field_name} decimal type does not have a required field precision")
+            if "scale" not in avro_field:
+                raise ValueError(f"{field_name} decimal type does not have a required field scale")
+            max_whole_number_range = avro_field["precision"] - avro_field["scale"]
+            decimal_range = avro_field["scale"]
+
+            # This regex looks like a mess, but it is validation for at least one whole number and optional fractional numbers
+            # For example: ^-?\d{1,5}(?:\.\d{1,3})?$ would accept 12345.123 and 123456.12345 would  be rejected
+            return {"type": "string", "pattern": f"^-?\\d{{{1,max_whole_number_range}}}(?:\\.\\d{1,decimal_range})?$"}
```

Comment:
why do we care about encoding this in the schema? Is it to keep track of the precision?

Reply:
I don't think it's encoding but rather because the avro's decimal logical type has the concept of precision and scale which tracks how many decimals. We can add that layer of validation with the pattern